### ISSUE TITLE
sender see no balance after a transfer (tapret1st)

### ISSUE THREAD

**Issue by nicbus** [RGB-WG/rgb#173]
Using commit `f9813b7` plus a cargo update to include rgb-std #193 I face an issue.

This is what I'm doing (wallet 1 is the issuer/sender, wallet 2 is the recipient):
- issue a NIA asset on wallet 1
- send some assets from wallet 1 to wallet 2

All operations are carried out using `tapret1st` closing method. The same issue doesn't manifest itself when using `opret1st`.

After the 1st transfer, the sender doesn't see the assets allocated as RGB change.

These are the sender UTXOs after the transfer:
```
HeightAmount, ṩ    Outpoint
Loadingdescriptor from wallet wallet_1 ... success

Wallettotal balance: 0 ṩ
```

This is the sender's contract state after the transfer:
```
Global:
  spec := (ticker=("USDT"), name=("USD Tether"), details=~, precision=0)
  terms := (text=("demo NIA asset"), media=~)
  issuedSupply := (2000)

Owned:
  assetOwner:
```

**Comment by dr-orlovsky** [comment]
Looks like issue with detecting outputs containing tweaks... Can you please include the sender data directory?

**Comment by nicbus** [comment]
Here is the whole sender's data directory.

[sender_data.tar.gz](https://github.com/RGB-WG/rgb/files/15036771/sender_data.tar.gz)

**Comment by dr-orlovsky** [comment]
Ok, this was trivial: tapret tweak was added to the wallet, but the wallet itself wasn't saved to the disk. Fixed in https://github.com/RGB-WG/rgb/pull/175

**Comment by nicbus** [comment]
Re-tested using branch `v0.11` (commit `7aa7361`).

The sender now fails to prepare the transfer with:
```
thread 'main' panicked at .../rgb/src/runtime.rs:213:31:
unable to save wallet data: Toml(Error { inner: KeyNotString })
```

**Comment by dr-orlovsky** [comment]
Ok, this means my fix from https://github.com/RGB-WG/rgb/commit/f9813b780bf0b0295b13df23d5017565840d3994#diff-f643199b78cd917deee88c751028d85bc710961b6f7b874f3c99cf6935fd68b0L109-L118 doesn't actually work.

Finding another way to save using serde in a feature-gate aware way with no `cfg_eval` involved...

**Comment by dr-orlovsky** [comment]
With the new fix from https://github.com/RGB-WG/rgb/pull/175/commits/ac43eb2709fe06cb102b7083ea6205634573f2df using https://github.com/BP-WG/bp-core/pull/87 and https://github.com/BP-WG/bp-std/pull/24 this should be working now

**Comment by nicbus** [comment]
Using commit `ac43eb2` plus a cargo update, I get this error when trying to create a wallet:
```
error: unable to serialize wallet cache as YAML file - can only flatten structs and maps (got a string)
```

**Comment by nicbus** [comment]
moreover, I noticed the exit status of the command is `0` even if the reported error gets output and I think this should be fixed

**Comment by dr-orlovsky** [comment]
Try with e05f6c75a1a150851a4b08fcec74f02230de6bb1 - just do `cargo update`. The exit status is separate issue and I will work on it elsewhere

**Comment by nicbus** [comment]
Wallet creation now succeeds, but I get the error when syncing the wallet (`utxos --sync`) after receiving some funds:
```
Syncing keychain 0 .......... keychain 1 .......... keychain 9 .......... keychain 10 ........... success
Error: unable to serialize wallet cache as YAML file - can only flatten structs and maps (got a string)
```

**Comment by nicbus** [comment]
After including a fix in `bp-std` (commit `68ca949a`), wallet syncing now succeeds, but I get another error when syncing the sender (wallet 1) after the transfer (after the tx has been broadcast and confirmed, before accepting):
```
Loading descriptor from wallet wallet_1 ... Error: unable to parse TOML file - TOML parse error at line 7, column 1
  |
7 | "&10/1" = "^k$0ej@X7L@NTPs0zL$(MRC_`!3rCMxVn8)b0OE|00"
  | ^^^^^^^
invalid type: string "&10/1", expected a borrowed string
```

**Comment by nicbus** [comment]
Including fixes for bp-std (`6741b11d`) and rgb-core (`ff275566`) the issue is solved.
